funding/manager.go

```diff
@@ -4131,7 +4124,7 @@ func (f *Manager) handleInitFundingMsg(msg *InitFundingMsg) {
 		remoteMaxHtlcs:    maxHtlcs,
 		remoteChanReserve: chanReserve,
 		maxLocalCsv:       maxCSV,
-		channelType:       msg.ChannelType,
+		channelType:       chanType,
```

Comment:
Looks like this is the only line needed for the fix? Since after `implicitNegotiateCommitmentType`, `chanType` won't be nil.

Reply:
Unfortunately, it's not that simple.

Changing this value to `chanType` causes many other test cases to fail.  It makes us expect the peer to use explicit negotiation in `handleFundingAccept`, even if the peer never signaled the necessary feature bit.  And so we incorrectly fail the channel.

Looking at previous changes to the negotiation function, it seems to me that no one has really understood what the function is doing, and has instead applied bandaids to fix urgent bugs, while introducing new more subtle bugs.

I hope this PR helps fix the root problem once and for all.